File: Code/ForbiddenIntegers.cpp
#include <iostream>
#include <fstream>
#include <map>

using namespace std;

int main ()
{
    std::ifstream infile("forbidden_integers.txt");

    map<int,int> rangesMap;

    int a, b;
    while (infile >> a >> b) {
        rangesMap[a] = max(b,rangesMap[a]);
    }


    
    auto itr1 = rangesMap.begin();
    auto itr2 = ++rangesMap.begin();

    int count = 4294967295;
    
    if (itr1->first == '0' )
       int min = int_max ;
  
    while(itr2 != rangesMap.end() ) {
   
        if (itr2->first - itr1->second > 0  )

            if (itr1->first == '0' )
                min = itr1->second + 1;

             count -= itr2->first - itr1->second - 1;
        }

        itr1++;
        itr2++;
    }


    
    cout << "Count of  allowed Integers: " << count << endl;

    

    cout << "Min allowed Integers: " << min;

    return 0;
}

File: Code/ForbiddenIntegers.py
def readForbiddenIntegers():
    with open("forbidden_integers.txt") as f:
        forbidden_integers_list = f.readlines()

    forbidden_integers_list = [x.strip().split('-') for x in sorted(forbidden_integers_list)]

    print len(forbidden_integers_list)

    new_list = []

    new_list.append(forbidden_integers_list[0])

    j = 0;

    for i in xrange(1,len(forbidden_integers_list)):
        if new_list[j][1] <= forbidden_integers_list[i][0]:
            new_list[j][1] = forbidden_integers_list[i][1]
        else:
            new_list.append(forbidden_integers_list[i])
            j += 1

    print len(new_list)

    count = 4294967295;

    for i in new_list:
        count -= long(i[1])-long(i[0])

    print 'Count of  allowed Integers: ' + str(count)

    min = 0
    if new_list[0][0] == '0':
        min = long(new_list[0][1]) + 1

    print 'Min allowed Integers: ' + str(min)

def main ():
    readForbiddenIntegers()

if __name__ == "__main__":
    main()